sdk/src/main/java/com/amplitude/experiment/storage/Cache.kt

```diff
@@ -53,8 +55,20 @@ internal class LoadStoreCache<V>(
         putAll(values)
     }
 
-    fun store(values: Map<String, V> = cache) = synchronized(cache) {
-        storage.put(namespace, JSONObject(values).toString())
+    fun store(values: MutableMap<String, V> = cache) = synchronized(cache) {
+        val stringValues = values.mapNotNull { entry ->
+            try {
+                val value = transformStringFromV(entry.value)
```

Comment:
To make this more generic, add another transformer argument to the constructor and inject the right transformer function rather than using type inference to decide. Like what is done with the deserialization.

E.g.

```
internal class LoadStoreCache<V>(
    private val namespace: String,
    private val storage: Storage,
    private val decoder: ((value: String) -> V?)
    private val encoder: ((value: V) -> String)
) {